File: groupAna.java
class groupAna {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> map = new HashMap<>();
        for(String str:strs){
            char[] chArr = str.toCharArray();
            Arrays.sort(chArr);
            String sorted = String.valueOf(chArr);
            if(!map.containsKey(sorted)) {
                map.put(sorted,new ArrayList<>());
            }
            map.get(sorted).add(str);
        }
        return new ArrayList<>(map.values());
    }
}

File: GroupAna.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict1 = {}
        for item in strs:
            list1 = tuple(sorted(item))
            if list1 in dict1:
                l1 = []
                l1 = dict1[list1]
                l1.append(item)
                dict1[list1] = l1
            else:
                l1 = []
                l1.append(item)
                dict1[list1] = l1
                            
        return dict1.values()